docs/protocol/specifications/validation.md

```diff
@@ -294,55 +131,47 @@ Note: requirement for the input and output sums to be below 2<sup>63</sup> impli
 
 **Algorithm:**
 
-1. Let S be the input blockchain state.
-2. For each transaction in the block:
-    1. [Apply the transaction](#apply-transaction) to S, yielding a new state S′.
-    2. Replace S with S′.
-3. Replace the block header in S with the input block header, yielding a new state S′.
-4. Replace S with S′.
-5. Remove elements of the issuance memory in S where the expiration timestamp is less than the block’s timestamp, yielding a new state S′.
-6. Return S′.
+1. Evaluate the [consensus program](blockchain.md#block-header):
+    1. [Create a VM 1](vm1.md#vm-state) with initial state and expansion flag set to `false`.
+    2. [Prepare VM](vm1.md#prepare-vm) with program arguments from the block witness.
+    4. Set the VM’s program to the consensus program as specified by the blockchain state’s block header.
+    5. Execute [Verify Predicate](vm1.md#verify-predicate) operation. If it fails, halt and return false.
+2. [Validate the block](blockchain.md#block-header-validation) with “previous block header” set to the block header in the current blockchain state; if invalid, halt and return blockchain state unchanged.
+3. Let `S` be the input blockchain state.
+4. For each transaction in the block, in order:
+    1. [Apply the transaction](#apply-transaction) using the input block’s header to blockchain state `S`, yielding a new state `S′`.
+    2. If transaction failed to be applied (did not change blockchain state), halt and return the input blockchain state unchanged.
+    3. Test that [assets merkle root](blockchain.md#assets-merkle-root) of `S′` is equal to the assets merkle root declared in the block commitment; if not, halt and return blockchain state unchanged.
```

Comment:
OMGZZZ. You are 100% correct, how did this exist for 5 months without being noticed? 

Fixed: dea297d